plugins.md

```diff
@@ -0,0 +1,218 @@
+  useful if your plugin only works on some versions of a module, or if you are
+  patching internal mechanisms that are specific to a certain range of versions.
+* `patch`: A function describing how the given file should be patched. It will
+  be passed two arguments: the object exported from the file, and an object that
+  exposes functions for creating spans and propagating context (see
+  [What the API provides](#what-the-api-provides) for more details).
+
+A plugin simply exports a list of patch objects.
+
+For example, here's what a plugin for `express` might export:
+
+```js
+// express 4.x
+
+function patchModuleRoot(expressModule, api) {
+  // ...
+}
+
+module.exports = [
+  { file: '', versions: '4.x', patch: patchModuleRoot }
+];
+```
+
+In most cases, it should be sufficient to patch just the root (public exports)
+of the module, in which case the `file` field can set to `''` or omitted. Based
+on how the module being patched is implemented, however, it may be necessary to
+patch other parts of the module as well.
+
+### Enabling plugins in the Trace Agent
+
+Developers wishing to trace their applications may specify, in the
+[configuration][config-js] object passed to the trace agent, a `plugins` field
+with a key-value pair (module name, path to plugin). For
+example, to patch just `express` with a plugin named `TODO`:
+
+require('@google-cloud/trace-agent').start({
+  config: {
+    express: 'TODO'
+  }
+})
+
+At this time, the path to the plugin may either be a module name (`'TODO'`) or
+an absolute path to the plugin (`__dirname + '/TODO'`).
+
+## What the API provides
+
+The `api` object, in short, provides functions that facilitate the following:
+
+- Creating trace spans and add labels to them.
+- Getting information about how the trace agent was configured in the
+  current application.
+- Parsing and serializing trace contexts for the sake of propagating them over
+  the network.
+- Binding callbacks and event emitters with a CLS namespace in order to
+  propagate trace contexts across asynchronous boundaries.
+
+### Trace Spans
+
+These functions provide the capability to create trace spans, add labels to
+them, and close them. `transaction` and `childSpan` are instances of
+`Transaction` and `ChildSpan`, respectively.
+
+#### `api.createTransaction(options)`
+* `options`: [`TraceOptions`](trace-span-options)
+* Returns `Transaction`
+
+Creates and returns a new `Transaction` object. A `Transaction` object
+represents a root-level trace span, and exposes functions for adding labels,
+ending the span, and serializing the current trace context. Note that the
+underlying root span is started when this function is called.
+
+This function consults the trace agent's tracing policy to determine whether a
+trace span should actually be started or not. If the tracing policy prohibits
+a new span to be created under the current circumstances, this function will
+return `null` instead.
```

Comment:
Maybe we should consider returning a dummy `Transaction` with noop operations so that users don't need to embed `null` checks or worry about sampling (they should still know it may occur).